Jenkinsfile

```diff
@@ -0,0 +1,71 @@
+pipeline {
+    agent any
+
+    stages {
+
+        stage('Setting version') {
+            steps {
+                sh "./gradlew makeVersionFile"
+                script {
+                    def version = readFile('build/version')
+                    currentBuild.displayName = "${version} (#${env.BUILD_NUMBER})"
+                }
+            }
+        }
+
+        stage("Assemble") {
+            steps {
+                sh "./gradlew clean assemble"
```

Comment:
Can we avoid doing a `clean`? That would speed up the builds, because it can take advantage of build cache.
It is possible we cannot, because artifacts left behind by an earlier build mess with later builds. In that case the buildprocess is broken, and we should look into it at some point.

Reply:
I think we can, but I would suggest doing this anyway for PR and master builds to make sure we have no inconsistent state. WDYT?

Reply:
Fine by me, sometimes the effects from this breaking are quite subtle and not easily reproducible. Better to be safe than sorry in that case.